### PR TITLE
Port to pika 0.8.0

### PR DESCRIPTION
**Pull request by msimberg**
- [x] update pika spack package
- [x] add version constraint to `CMakeLists.txt`

### CODE REVIEW

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try-

**Comment by msimberg** [comment]
bors try

**Comment by rasolca** [comment]
bors try-

**Comment by rasolca** [comment]
bors try

**Comment by albestro** [comment]
Working on #645 with @rasolca we realised that probably we can take the chance in this PR, which is going to rebuild the base image, to remove also this line from the DLAF package.

https://github.com/eth-cscs/DLA-Future/blob/89b2e118ef6ec502bd54e3dd12e2b57c287b29d1/spack/packages/dla-future/package.py#L103

File: spack/packages/dla-future/package.py [hunk 1]
```diff
@@ -100,7 +100,6 @@ def cmake_args(self):
             args.append(self.define("BUILD_TESTING", True))
             args.append(self.define("DLAF_BUILD_TESTING", True))
             args.append(self.define("DLAF_CI_RUNNER_USES_MPIRUN", True))
-            args.append(self.define("MPIEXEC_EXECUTABLE", "srun"))
```
Comment (msimberg) [inline spack/packages/dla-future/package.py:103]: @albestro just like this?

Comment (albestro) [reply spack/packages/dla-future/package.py:103]: Yep, thanks!

**Comment by msimberg** [comment]
bors try

**Comment by rasolca** [comment]
bors try-

**Comment by rasolca** [comment]
bors try

**Comment by msimberg** [comment]
I think this is currently what is blocking the HIP builds from completing: https://github.com/spack/spack/pull/32025. Umpire requires CMake < 3.21 and pika requires CMake >= 3.22.

**Comment by rasolca** [comment]
What if we temporarly disable the AMD gpu pipeline until we can make it work again?

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Pull request merged**